lib/web/ui/app.go

```diff
@@ -68,6 +69,10 @@ func MakeApps(c MakeAppsConfig) []App {
 		fqdn := AssembleAppFQDN(c.LocalClusterName, c.LocalProxyDNSName, c.AppClusterName, teleApp)
 		labels := []Label{}
 		for name, value := range teleApp.GetAllLabels() {
+			if strings.HasPrefix(name, "teleport.internal/") {
+				continue
+			}
```

Comment:
Instead of repeating the same filtering logic in all locations and writing tests for each, why not create a function that removes labels from the map returned by `GetAllLabels`? That way we'd only need to test the new function rather than adding tests for all the places that require this logic.

```go
func removeHiddenLabels(m map[string]string) map[string]string {
	for name := range m {
		if strings.HasPrefix(name, "teleport.internal/") {
			delete(m, name)
		}
	}
	return m // return map so it can be used directly in for loops
}

// ... then 
for name, value := range removeHiddenLabels(teleApp.GetAllLabels()) {
```

This would isolate all testing to this new func - no need to test all places where the labels are hidden. Also makes it much easier to add new scenarios for hidden labels.

Reply:
I agree that function would be good, I could go even deeper because there is no need to repeat logic appending Labels to slice. It looks exactly the same everywhere. 

About tests, I actually would like to keep them. I would even love to add more unit tests for those functions, as they are not tested that way. Unless there is a reason not to.